ppl/archive/index.go

```diff
@@ -0,0 +1,106 @@
+package archive
+
+import (
+	"context"
+	"runtime"
+
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/ppl/archive/chunk"
+	"github.com/brimsec/zq/ppl/archive/index"
+	"github.com/brimsec/zq/zio/zngio"
+	"github.com/brimsec/zq/zng/resolver"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+)
+
+// EnsureIndices walks through the entirety of an Achive's chunks ensuring that
+// all chunk indices are update-to-date with an Archive's IndexDefs.
+func EnsureIndices(ctx context.Context, ark *Archive, progress chan<- string) error {
```

Comment:
Isn't this exporting an private type?

Reply:
You're correct - I mistakenly thought you were going to make the `progress` type exported as well.
Instead of my code change, would you add a comment explaining that the index paths are sent on the channel?